packages/block-library/src/list-item/edit.js

```diff
@@ -1,47 +1,242 @@
+		// whose descendants are the children of the previous sibling, followed by the indented block.
+		replaceBlocks(
+			[ previousSiblingId, clientId ],
+			[
+				createListItem(
+					previousSibling.attributes,
+					previousSiblingListAttributes,
+					[ ...previousSiblingChildren, newBlock ]
+				),
+			]
+		);
+
+		// Restore the selection state.
+		selectionChange(
+			newBlock.clientId,
+			selectionEnd.attributeKey,
+			selectionEnd.clientId === selectionStart.clientId
+				? selectionStart.offset
+				: selectionEnd.offset,
+			selectionEnd.offset
+		);
+	}, [ clientId ] );
+}
+
+function useOutdentListItem( clientId ) {
+	const { replaceBlocks, selectionChange } = useDispatch( blockEditorStore );
+	return useCallback( () => {
+		const {
+			getBlockRootClientId,
+			getBlockAttributes,
+			getBlock,
+			getBlockIndex,
+			getSelectionStart,
+			getSelectionEnd,
+		} = select( blockEditorStore );
+		const selectionStart = getSelectionStart();
+		const selectionEnd = getSelectionEnd();
+
+		const listParentId = getBlockRootClientId( clientId );
+		const listAttributes = getBlockAttributes( listParentId );
+		const listItemParentId = getBlockRootClientId( listParentId );
+		const listItemParentAttributes = getBlockAttributes( listItemParentId );
+
+		const index = getBlockIndex( clientId );
+		const siblingBlocks = getBlock( listParentId ).innerBlocks;
+		const previousSiblings = siblingBlocks.slice( 0, index );
+		const afterSiblings = siblingBlocks.slice( index + 1 );
+
+		// Create a new parent list item block with just the siblings
+		// that existed before the child item being outdent.
+		const newListItemParent = createListItem(
+			listItemParentAttributes,
+			listAttributes,
+			previousSiblings
+		);
+
+		const block = getBlock( clientId );
+		const childList = first( block.innerBlocks );
+		const childItems = childList?.innerBlocks || [];
+		const hasChildItems = !! childItems.length;
+
+		// Create a new list item block whose attributes are equal to the
+		// block being outdent and whose children are the children that it had (if any)
+		// followed by the siblings that existed after it.
+		const newItem = createListItem(
+			block.attributes,
+			hasChildItems ? childList.attributes : listAttributes,
+			[ ...childItems, ...afterSiblings ]
+		);
+
+		// Replace the parent list item block, with a new block containing
+		// the previous siblings, followed by another block containing after siblings
+		// in relation to the block being outdent.
+		replaceBlocks( [ listItemParentId ], [ newListItemParent, newItem ] );
+
+		// Restore the selection state.
+		selectionChange(
+			newItem.clientId,
+			selectionEnd.attributeKey,
+			selectionEnd.clientId === selectionStart.clientId
+				? selectionStart.offset
+				: selectionEnd.offset,
+			selectionEnd.offset
+		);
+	}, [ clientId ] );
+}
+
+function IndentUI( { clientId } ) {
+	const { canOutdent, canIndent } = useSelect(
```

Comment:
Makes me wonder whether these should be part of the `useIndentListItem` (return a boolean and a function)